parser/ast/dml.go

```diff
@@ -1042,6 +1042,17 @@ type CommonTableExpression struct {
 	Name        model.CIStr
 	Query       *SubqueryExpr
 	ColNameList []model.CIStr
+	IsRecursive bool
+}
+
+// Restore implement it Before using
+func (c CommonTableExpression) Restore(ctx *format.RestoreCtx) error {
+	panic("implement me")
```

Comment:
It shouldn't panic here.

Reply:
What is the recommand code here, i implement it because of compile check.But is is not use in the program.And i add note on it.

Reply:
if len(cte.ColNameList) > 0 {
			ctx.WritePlain(" (")
			for j, name := range cte.ColNameList {
				if j != 0 {
					ctx.WritePlain(", ")
				}
				ctx.WriteName(name.String())
			}
			ctx.WritePlain(")")
		}
		ctx.WriteKeyWord(" AS ")
		err := cte.Query.Restore(ctx)
		if err != nil {
			return err
		}


We can move some code from (n *WithClause) Restore

Reply:
Get